public/app/app.ts

```diff
@@ -121,6 +121,10 @@ export class GrafanaApp {
       setPanelDataErrorView(PanelDataErrorView);
       setLocationSrv(locationService);
       setTimeZoneResolver(() => config.bootData.user.timezone);
+
+      // We must wait for translations to load because some preloaded store state requires translating
+      await initI18nPromise;
```

Comment:
Alternatively, maybe we could await this further down like we used to, and dispatch an action later on (but before any react rendering) to translate/initialize the nav?